### PR TITLE
add custom capo readme

### PR DESCRIPTION
**Pull request by Gal-Zaidman**

### CODE REVIEW

File: docs/custom-capo.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Creating a custom cluster api provider ovirt and adding it to the cluster
+
+When we develop the CAPO its usually helpful to test it against a running cluster.
+The following doc explains how to do it.
+
+## Replacing the docker image
```
Comment (dougsland) [inline docs/custom-capo.md:6]: I would replace "docker" as container.

File: docs/custom-capo.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Creating a custom cluster api provider ovirt and adding it to the cluster
+
+When we develop the CAPO its usually helpful to test it against a running cluster.
```
Comment (dougsland) [inline docs/custom-capo.md:3]: What's CAPO? :)  I am assuming is: Cluster API Provider OVirt (CAPO) 

I would than use: When we develop the Cluster API Provider oVirt (CAPO) its ....

File: docs/custom-capo.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+The following doc explains how to do it.
+
+## Replacing the docker image
+
+1.Lunch a cluster:
+
+Start an Openshift cluster on top of oVirt as usual with the required release image.
```
Comment (dougsland) [inline docs/custom-capo.md:10]: I would like a note here like:

For more info about installing and launching a new cluster using oVirt check:
 - [Installer Provisioned Infrastructure (IPI)](https://github.com/openshift/installer/blob/master/docs/user/ovirt/install_ipi.md)
 - [User Provisioned Infrastructure (UPI)](https://github.com/openshift/installer/blob/master/docs/user/ovirt/install_upi.md)

File: docs/custom-capo.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Creating a custom cluster api provider ovirt and adding it to the cluster
+
+When we develop the CAPO its usually helpful to test it against a running cluster.
+The following doc explains how to do it.
+
+## Replacing the docker image
+
+1.Lunch a cluster:
```
Comment (dougsland) [inline docs/custom-capo.md:8]: Launch

File: docs/custom-capo.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Start an Openshift cluster on top of oVirt as usual with the required release image.
+
+2. Create the docker image:
+
+** replace ${username} and ${tag} with the correct values
+```bash
+podman login quay.io
```
Comment (dougsland) [inline docs/custom-capo.md:16]: I would recommend use $ in front of commands, specially this shows podman doesn't require root as others solutions like docker.

```
$ podman login quay.io
$ podman build -t quay.io/${username}/cluster-api-provider-ovirt:${tag} .
$ podman push quay.io/${username}/cluster-api-provider-ovirt:${tag}
```

File: docs/custom-capo.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+## Replacing the docker image
+
+1.Lunch a cluster:
+
+Start an Openshift cluster on top of oVirt as usual with the required release image.
+
+2. Create the docker image:
+
+** replace ${username} and ${tag} with the correct values
```
Comment (dougsland) [inline docs/custom-capo.md:14]: ** doesn't show special char in Markdown. 
I would move this comment below the podman commands below, like:

**NOTE**:
${username} and ${tag} make be replaced with the the proper quay.io username and cluster-api-provider-ovirt tag.

File: docs/custom-capo.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+EOF
+)"
+```
+
+Scale down cluster-version-operator:
+```bash
+oc scale --replicas=0 deploy/cluster-version-operator -n openshift-cluster-version
+```
+
+IMPORTANT: This apprach disables cluster-version-operator completly, whereas previous only tells it to not manage a kube-scheduler-operator!
```
Comment (dougsland) [inline docs/custom-capo.md:43]: **IMPORTANT**:
This approach

File: docs/custom-capo.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+4. Replace the CAPO image
+
+```bash
+oc -n openshift-machine-api edit configmap/machine-api-operator-images
```
Comment (dougsland) [inline docs/custom-capo.md:48]: $ oc

File: docs/custom-capo.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+```
+and replace clusterAPIControllerOvirt with the pull (by digest) of your image
```
Comment (dougsland) [inline docs/custom-capo.md:50]: Users and devel will ask: How can I find the digest from the my image? Can we add an example here?

File: docs/custom-capo.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+5. verify that the image has been replaced:
+
+```bash
+oc -n openshift-machine-api get pods
```
Comment (dougsland) [inline docs/custom-capo.md:55]: $ oc

Comment (dougsland) [inline docs/custom-capo.md:55]: I would copy/past a real example of the command

File: docs/custom-capo.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+verify that pod pod/machine-api-controllers-***-*** has been updated
```
Comment (dougsland) [inline docs/custom-capo.md:58]: maybe: pod/machine-api-controllers-**VERSION**-**RELEASE** ?

**Review by dougsland — CHANGES_REQUESTED** [review]
Thanks for sharing your work in progress doc Gal. I have few suggestions.

File: docs/custom-capo.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
+# Creating a custom cluster api provider ovirt and adding it to the cluster
```
Comment (dougsland) [inline docs/custom-capo.md:1]: I would replace with: Creating a custom Cluster API Provider for oVirt

File: docs/custom-capo.md (hunk 11 above)
Comment (Gal-Zaidman) [reply docs/custom-capo.md:75]: The pod has some randome chars after the name :)

**Comment by dougsland** [comment]
documentation only

**Pull request merged**